hoot-services/src/main/java/hoot/services/controllers/auth/OAuth1Resource.java

```diff
@@ -0,0 +1,181 @@
+@Path("")
+@Transactional
+public class OAuth1Resource {
+    private static final Logger logger = LoggerFactory.getLogger(OAuth1Resource.class);
+    @Autowired
+    private OAuthRestTemplate oauthRestTemplate;
+    @Autowired
+    private OAuthConsumerTokenServices tokenServices;
+    @Autowired
+    private UserManager userManager;
+
+
+    @GET
+    @Path("/oauth1/verify")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response verify(@QueryParam("oauth_token") String oauth_token, @QueryParam("oauth_verifier") String oauth_verifier) {
+        OAuthConsumerSupport s = oauthRestTemplate.getSupport();
+        ProtectedResourceDetails r = oauthRestTemplate.getResource();
+        HttpSession sess = getSession();
+        if (sess == null) {
+            return Response.status(401).build();
+        }
+        sess.setMaxInactiveInterval(31536000);
+        OAuthConsumerToken requestToken = tokenServices.getToken(r.getId());
+        if (requestToken == null) {
+            return Response.status(401).build();
+        }
+        OAuthConsumerToken accessToken = null;
+        try {
+            accessToken = s.getAccessToken(r /* <== Specifying this was important [!] */, requestToken, oauth_verifier);
+        } catch (Exception e) {
+            // Pass.
+        }
+        if (accessToken == null) {
+            return Response.status(401).build();
+        }
+        // Save this access token to our HTTP Session based token services:
+        tokenServices.storeToken(r.getId(), accessToken);
+        // Save this access token to our security context, used by
+        // OAuthClientHttpRequestFactory to authorize
+        // outbound requests.
+        OAuthSecurityContext context = OAuthSecurityContextHolder.getContext();
+        context.getAccessTokens().put(r.getId(), accessToken);
+
+        String response = oauthRestTemplate.getForObject("https://api.openstreetmap.org/api/0.6/user/details", String.class);
+        Users user;
+        try {
+            user = userManager.upsert(response, accessToken, sess.getId());
+        } catch (InvalidUserProfileException | SAXException | IOException | ParserConfigurationException e) {
+            logger.error("Failed to read user profile from oauth provider", e);
+            return Response.status(502).build();
+        } catch (Exception e) {
+            logger.error("Failed to store user during oauth verification", e);
+            return Response.status(500).build();
+        }
+
+        return Response.status(200).entity(user).type(MediaType.APPLICATION_JSON).build();
+    }
+
+    @GET
+    @Path("/oauth1/request")
+    @Produces(MediaType.TEXT_PLAIN)
+    public Response request() {
+        OAuthConsumerSupport s = oauthRestTemplate.getSupport();
+        ProtectedResourceDetails r = oauthRestTemplate.getResource();
+        OAuthConsumerToken requestToken = null;
+        try {
+            requestToken = s.getUnauthorizedRequestToken(r, "http://localhost:8080/");
```

Comment:
Will this have to match the deployed production url?

Reply:
yes

Reply:
We decided to put the value in hoot-services.conf